labelCloud/view/startup_dialog.py

```diff
@@ -0,0 +1,228 @@
+
+        self.main_layout = QVBoxLayout()
+
+        row_buttons = QHBoxLayout()
+        self.button_object_detection = QPushButton(text=LabelingMode.OBJECT_DETECTION.title().replace("_", " "))
+        self.button_object_detection.setCheckable(True)
+        self.button_semantic_segmentation = QPushButton(text=LabelingMode.SEMANTIC_SEGMENTATION.title().replace("_", " "))
+        self.button_semantic_segmentation.setCheckable(True)
+
+        row_buttons.addWidget(self.button_object_detection)
+        row_buttons.addWidget(self.button_semantic_segmentation)
+        self.main_layout.addLayout(row_buttons)
+        self.label_name_validator = LabelNameValidator()
+
+        self.load_class_labels(self.main_layout)
+
+        self.button_add_label = QPushButton(text="Add new label")
+        self.button_add_label.clicked.connect(self.add_label_row)
+        self.main_layout.addWidget(self.button_add_label)
+
+        self.buttonBox = QDialogButtonBox(QDialogButtonBox.Save)
+        self.buttonBox.accepted.connect(self.accept)
+        self.buttonBox.rejected.connect(self.reject)
+        self.button_object_detection.clicked.connect(self.switch_to_object_detection_mode)
+        self.button_semantic_segmentation.clicked.connect(self.switch_to_semantic_segmentation_mode)
+        self.main_layout.addWidget(self.buttonBox)
+        self.main_layout.setAlignment(Qt.AlignTop)
+        self.setLayout(self.main_layout)
+
+
+        self.delete_buttons.buttonClicked.connect(self.delete_label_row)
+
+    def switch_to_object_detection_mode(self) -> None:
+        self.button_object_detection.setChecked(True)
+        self.button_semantic_segmentation.setChecked(False)
+
+    def switch_to_semantic_segmentation_mode(self) -> None:
+        self.button_object_detection.setChecked(False)
+        self.button_semantic_segmentation.setChecked(True)
+
+    def get_labeling_mode(self) -> LabelingMode:
+        if self.button_object_detection.isChecked():
+            return LabelingMode.OBJECT_DETECTION
+        return LabelingMode.SEMANTIC_SEGMENTATION
+
+    def load_class_labels(self, main_layout: QVBoxLayout) -> None:
+
+        self.class_labels = QVBoxLayout()
+        self.delete_buttons = QButtonGroup()
+        self.delete_button_hash: List[int] = []
+
+        for class_label in LabelConfig().classes:
+            row_label = QHBoxLayout()
+            # label id
+            label_id = QSpinBox()
+            label_id.setMinimum(0)
+            label_id.setMaximum(255)
+            label_id.setValue(class_label.id)
+            # label name
+            label_name = QLineEdit(class_label.name)
+            
+            label_name.setValidator(self.label_name_validator)
+            # label color
+            label_color = ColorButton(color=rgb_to_hex(class_label.color))
+            # delete button
+            label_delete = QPushButton(text="X")
+            self.delete_buttons.addButton(label_delete, hash(label_delete))
```

Comment:
I always get the following exception here:
```
self.delete_buttons.addButton(label_delete, hash(label_delete))
OverflowError: argument 2 overflowed: value must be in the range -2147483648 to 2147483647
```

What is the background for using the hashes here?

Reply:
I think the hash can be at times to high for the id. I changed so that the `button.id` is now directly set as the row index, which also removes the necessity for the hash list.

But otherwise the ButtonGroup seems like a nice fit for this kind of problem.